src/server.h

```diff
@@ -969,26 +970,12 @@ typedef struct replBufBlock {
     char buf[];
 } replBufBlock;
 
-typedef struct dbDictState {
-    list *rehashing;                       /* List of dictionaries in this DB that are currently rehashing. */
-    int resize_cursor;                     /* Cron job uses this cursor to gradually resize dictionaries (only used for cluster-enabled). */
-    int non_empty_slots;                   /* The number of non-empty slots. */
-    unsigned long long key_count;          /* Total number of keys in this DB. */
-    unsigned long long bucket_count;       /* Total number of buckets in this DB across dictionaries (only used for cluster-enabled). */
-    unsigned long long *slot_size_index;   /* Binary indexed tree (BIT) that describes cumulative key frequencies up until given slot. */
-} dbDictState;
-
-typedef enum dbKeyType {
-    DB_MAIN,
-    DB_EXPIRES
-} dbKeyType;
-
 /* Redis database representation. There are multiple databases identified
  * by integers from 0 (the default database) up to the max configured
  * database. The database number is the 'id' field in the structure. */
 typedef struct redisDb {
-    dict **dict;                /* The keyspace for this DB */
-    dict **expires;             /* Timeout of keys with a timeout set */
+    dictarray *keys;            /* The keyspace for this DB */
+    dictarray *volatile_keys;   /* Timeout of keys with a timeout set */
```

Comment:
i think we wanna keep calling it `expires`

Reply:
i'm keeping this comment open for now because i think "volatile_keys" is a better name... let's see what others think

Reply:
i'm in favor of just keeping the same variable name we had in 7.2.
i.e. this PR will make the diff between 8.0 and 7.2 slightly simpler (compared to what's in unstable now)